### PR TITLE
[CI] Reflect changes made in LLPC's nightly configuration

### PR DESCRIPTION
**Pull request by vettoreldaniele**
The `+clang+assertions` configuration has been removed in https://github.com/GPUOpen-Drivers/llpc/pull/1561.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
Can one of the admins verify this patch?

File: .github/workflows/check-xgl-docker.yml [hunk 1]
```diff
@@ -17,8 +17,8 @@ jobs:
         host-os:             ["ubuntu-20.04"]
         base-image-template: ["gcr.io/stadia-open-source/amdvlk_%s%s:nightly"]
         config:              [Release]
-        feature-set:         ["+gcc", "+clang",
-                              "+gcc+assertions", "+clang+assertions"]
+        feature-set:         ["+gcc", "+gcc+assertions",
+                              "+clang"]
```
Comment (kuhar) [inline .github/workflows/check-xgl-docker.yml:21]: How about we add some sanitizer-enabled configurations?

Comment (kuhar) [reply .github/workflows/check-xgl-docker.yml:21]: AFAIK we should always build cache-creator now, so any non-coverage configuration should work.

**Comment by vettoreldaniele** [comment]
V1: Added `+clang+shadercache+ubsan+asan+assertions` configuration in place of `+clang+assertions`.

**Review by kuhar — COMMENTED** [review]
LGTM. We should force-push to confirm this is working once the LLPC change lands.

**Comment by kuhar** [comment]
@vettoreldaniele ping

**Comment by vettoreldaniele** [comment]
> @vettoreldaniele ping

The latest AMDVLK build failed (for unrelated reasons) so there are no images built yet for `+clang+shadercache+ubsan+asan+assertions`.

**Comment by kuhar** [comment]
> > @vettoreldaniele ping
> 
> The latest AMDVLK build failed (for unrelated reasons) so there are no images built yet for `+clang+shadercache+ubsan+asan+assertions`.

The previous one succeeded: https://github.com/GPUOpen-Drivers/llpc/runs/4437886178?check_suite_focus=true. We only upload images that pass all tests, so it should not matter that the very last one failed.

**Comment by kuhar** [comment]
> The previous one succeeded: https://github.com/GPUOpen-Drivers/llpc/runs/4437886178?check_suite_focus=true. We only upload images that pass all tests, so it should not matter that the very last one failed.

(unless submodule bump will cause it to fail again. We only bump submodules, not external dependencies, so this may be fine)

**Comment by vettoreldaniele** [comment]
> > > @vettoreldaniele ping
> > 
> > 
> > The latest AMDVLK build failed (for unrelated reasons) so there are no images built yet for `+clang+shadercache+ubsan+asan+assertions`.
> 
> The previous one succeeded: https://github.com/GPUOpen-Drivers/llpc/runs/4437886178?check_suite_focus=true. We only upload images that pass all tests, so it should not matter that the very last one failed.

It's not the right config (shadercache is missing).

**Comment by vettoreldaniele** [comment]
The UB sanitizer found UB in one of the tests. I will fix the tests and force push this again.

**Comment by vettoreldaniele** [comment]
V2: disabled ValidBlobOneEntry test because of UB. Being tracked in https://github.com/GPUOpen-Drivers/xgl/issues/143

**Review by kuhar — APPROVED** [review]

**Comment by vettoreldaniele** [comment]
V3: disable sanitizer configuration for XGL until we fix underlying issues.

**Review by kuhar — APPROVED** [review]

**Pull request merged**